### PR TITLE
add must_gather test for OCS4_7

### PR DESCRIPTION
**Pull request by OdedViner**
Signed-off-by: Oded Viner <oviner@redhat.com>

### CODE REVIEW

File: ocs_ci/ocs/must_gather/const_must_gather.py [hunk 1]
```diff
@@ -731,4 +731,10 @@
         "OTHERS": GATHER_COMMANDS_OTHERS + GATHER_COMMANDS_OTHERS_4_6,
         "OTHERS_EXTERNAL": GATHER_COMMANDS_OTHERS_EXTERNAL_4_6,
     },
+    4.7: {
+        "CEPH": GATHER_COMMANDS_CEPH + GATHER_COMMANDS_CEPH_4_6,
+        "JSON": GATHER_COMMANDS_JSON + GATHER_COMMANDS_JSON_4_6,
+        "OTHERS": GATHER_COMMANDS_OTHERS + GATHER_COMMANDS_OTHERS_4_6,
+        "OTHERS_EXTERNAL": GATHER_COMMANDS_OTHERS_EXTERNAL_4_6,
+    },
```
Comment (Avilir) [inline ocs_ci/ocs/must_gather/const_must_gather.py:739]: does it need to be 4_7 ?

File: ocs_ci/ocs/must_gather/const_must_gather.py [hunk 2]
```diff
@@ -731,4 +745,10 @@
         "OTHERS": GATHER_COMMANDS_OTHERS + GATHER_COMMANDS_OTHERS_4_6,
         "OTHERS_EXTERNAL": GATHER_COMMANDS_OTHERS_EXTERNAL_4_6,
     },
+    4.7: {
+        "CEPH": GATHER_COMMANDS_CEPH + GATHER_COMMANDS_CEPH_4_7,
+        "JSON": GATHER_COMMANDS_JSON + GATHER_COMMANDS_JSON_4_7,
+        "OTHERS": GATHER_COMMANDS_OTHERS + GATHER_COMMANDS_OTHERS_4_6,
+        "OTHERS_EXTERNAL": GATHER_COMMANDS_OTHERS_EXTERNAL_4_6,
```
Comment (PrasadDesala) [inline ocs_ci/ocs/must_gather/const_must_gather.py:752]: "_4_6", Should it be 4_7?

File: ocs_ci/ocs/must_gather/const_must_gather.py [hunk 3]
```diff
@@ -731,4 +745,10 @@
         "OTHERS": GATHER_COMMANDS_OTHERS + GATHER_COMMANDS_OTHERS_4_6,
         "OTHERS_EXTERNAL": GATHER_COMMANDS_OTHERS_EXTERNAL_4_6,
     },
+    4.7: {
```
Comment (PrasadDesala) [inline ocs_ci/ocs/must_gather/const_must_gather.py:748]: Are you also planning to make the new noobaa pod name changes in this PR?
Noobaa DB pod from 4.7 changed - from noobaa-db-0 to noobaa-db-pg-0

File: ocs_ci/ocs/must_gather/const_must_gather.py [hunk 4]
```diff
@@ -731,4 +777,10 @@
...
+        "CEPH": GATHER_COMMANDS_CEPH + GATHER_COMMANDS_CEPH_4_7,
+        "JSON": GATHER_COMMANDS_JSON + GATHER_COMMANDS_JSON_4_7,
+        "OTHERS": GATHER_COMMANDS_OTHERS + GATHER_COMMANDS_OTHERS_4_7,
+        "OTHERS_EXTERNAL": GATHER_COMMANDS_OTHERS_EXTERNAL_4_6,
```
Comment (PrasadDesala) [inline ocs_ci/ocs/must_gather/const_must_gather.py:784]: "_4_6", Should it be 4_7?
>>Should we change it for OTHERS_EXTERNAL too?

Comment (OdedViner) [reply ocs_ci/ocs/must_gather/const_must_gather.py:784]: @sidhant-agrawal

**Review by vasukulkarni — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]
@petr-balogh @clacroix12 @vasukulkarni I think this is a good example why we should have branches for OCS versions

**Comment by clacroix12** [comment]
> @petr-balogh @clacroix12 @vasukulkarni I think this is a good example why we should have branches for OCS versions

Agreed!

File: ocs_ci/ocs/must_gather/const_must_gather.py (hunk 4 above)
Comment (sidhant-agrawal) [reply ocs_ci/ocs/must_gather/const_must_gather.py:783]: apologies for the late reply.
We need to check if any new files added for Internal mode 4.7 is also valid for External mode 4.7
In External mode, they could have a slightly different name containing `external` keyword, or could be entirely absent.

If new files in 4.7 not applicable for external mode, then we can continue using 4.6 list

**Pull request merged**